packages/cursorless-engine/src/core/commandRunner/CommandRunner.ts

```diff
@@ -86,7 +90,21 @@ export class CommandRunner {
         usePrePhraseSnapshot,
       );
 
-      const action = this.actions[actionName];
+      const scopeHandlerFactory = new ScopeHandlerFactoryImpl(
+        this.languageDefinitions,
+      );
+      const markStageFactory = new MarkStageFactoryImpl(readableHatMap);
+      const modifierStageFactory = new ModifierStageFactoryImpl(
+        this.languageDefinitions,
+        scopeHandlerFactory,
+      );
+      const actions: ActionRecord = new Actions(
+        this.snippets,
+        this.rangeUpdater,
+        modifierStageFactory,
+      );
```

Comment:
We now construct all of these on the fly so that we can pass `readableHatMap` to the `DecoratedSymbolStage`.  I think constructing them on the fly will make our lives easier down the road as well, so that they can access dynamic things if necessary.  We're basically giving these components the singleton lifestyle

Note that I'm treating CommandRunner as part of the composition root, because it depends on everything and doesn't do too much fancy stuff.  We can prob try to split it up down the road to further isolate any complexity.

Maybe I should move it closer to `cursorlessEngine` to reflect is position as part of the composition root?

cc/ @auscompgeek